cmd/groups-auto-updater/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config/org"
+
+	userV1 "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+// Config holds the information of all group names and their configuration
+type Config struct {
+	Groups map[string]GroupConfig `json:"groups"`
+}
+
+func (c *Config) validate() error {
+	for group, config := range c.Groups {
+		if config.FromPeribolosConfig != nil {
```

Comment:
For instance, what is the use of this? In what case do we care to configure this but not configure _how_?